nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -154,6 +155,10 @@ def run(self, data=None, commit=None):
             for site in site_list:
                 for other_circuit_maint in circuit_maintenance_mapper[site.name]:
                     # Report failures for any time where a circuit will take an outage
+                    maintenance_list.append(circuit_maint)
```

Comment:
why we try to append  the `circuit_maint` from the outer `circuit_maintenances` loop, within the `site_list` loop. It looks counterintuitive to me. I would di is as soon as possible.
Also, seeing this, the `list` maybe better a `set`?

Reply:
I had some issues with sets, I like sets. But the list was able to handle it.